components/supertype/package.json

```diff
@@ -22,17 +32,19 @@
         "nyc": "15.1.0",
         "sinon": "15.0.1",
         "ts-node": "10.9.1",
+        "ttypescript": "^1.5.15",
         "typescript": "^4.9.5"
     },
     "scripts": {
         "lint": "eslint .",
-        "debug:tests": "mocha --inspect-brk test/animals.js",
+        "debug:tests": "mocha --inspect-brk test/animals.mjs",
         "debug:tstests": "mocha --inspect-brk test/typescript/*.js",
-        "test:ts": "mocha --require ts-node/register test/typescript/*",
+        "test:ts": "TS_NODE_PROJECT='./tsconfig.cjs.json' mocha --require ts-node/register test/typescript/*",
+        "test:ts:esm": "TS_NODE_PROJECT='./test/tsconfig.json' mocha --loader=ts-node/esm test/typescriptEsm/*",
```

Comment:
nit: may be we can also rename the `esm`  related `tconfig.json` file to be `tsconfig.esm.json`, to be consistent with name conventions.